src/realm/sync/noinst/client_reset.cpp

```diff
@@ -74,6 +74,15 @@ void transfer_group(const Transaction& group_src, Transaction& group_dst, util::
 {
     logger.debug("transfer_group, src size = %1, dst size = %2", group_src.size(), group_dst.size());
 
+    // Turn off the sync history tracking during state transfer since it will be thrown
+    // away immediately after anyways. This reduces the memory footprint of a client reset.
+    ClientReplication* client_repl = dynamic_cast<ClientReplication*>(group_dst.get_replication());
+    REALM_ASSERT_RELEASE(client_repl);
+    client_repl->set_short_circuit(true);
```

Comment:
`TempShortCircuitReplication` is a helper to do this.

Reply:
Just what I needed, thanks! Updated.